src/commands/build/index.js

```diff
@@ -35,10 +37,11 @@ class BuildCommand extends Command {
 
 // Netlify Build programmatic options
 BuildCommand.flags = {
-  dry: flags.boolean({
+  dry: flagsLib.boolean({
     description: 'Dry run: show instructions without running them',
+    default: false,
```

Comment:
I removed the casting of `Boolean(options.dry)` in favor of a default value (without a default value not passing `--dry` results in `flags.dry` being `undefined`)